### PR TITLE
Calculator:Milestone-4

### PR DESCRIPTION
**Pull request by jebitok-dev**
In this milestone I:-
- implemented the logic(operate.js and calculate.js) using big.js and the respective button names and numbers
- added calculate to App component

### CODE REVIEW

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      break;
+    case '*':
+      ans = x * y;
+      break;
+    case '/':
+      ans = x / y;
+      break;
+    case '%':
+      ans = (x % y) / 100;
+      break;
```
Comment (uche-inyama) [inline src/logic/operate.js]: case '%' should divide an input by 100.
```
   case '%':      
        ans = (x) / 100;      
  break;
```
For example: 95 % == .95

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import operate from './operate';
+
+function calculate(calculatorDataObj, buttonName) {
+  let { total, next, operation } = calculatorDataObj;
+  if (buttonName === 'AC') {
+    total = '';
+    next = '';
+    operation = '';
+  } else if (buttonName === '+/-') {
+    total = next * total * -1;
```
Comment (uche-inyama) [inline src/logic/calculate.js]: if button's name == "+/-", total and next should be multiplied by -1 to make it negative or positive. 

for example:
```

 if (buttonName === '+/-') {    
         return {      
           ...obj,     
              total: total * -1,     
             next: next * -1,    
           }; 
      }
```

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes required :recycle: 

### Hi Sharon Jebitok :slightly_smiling_face: 

This milestone is coming along fine, however,

- Update the case for percentage in logic/operate.js.
- Also, in logic/calculate.js reimplement the condition for '+/-'

 I have left an inline comment below.

#### Make these change then submit another form for a code review.

### Happy coding :man_technologist: 

#### ``` Uchechukwu Henry Inyama ```

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi @jebitok-dev , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**